### PR TITLE
Sujitu/create cartwith freetrial product

### PR DESCRIPTION
**Pull request by sujith0406**
…n modern purchases.

# Description

Updated the scenario for create cart to support free trial products in modern purchases.

## Related issue

NA

### CODE REVIEW

File: sdk/SdkSamples/Carts/CreateCart.cs [hunk 1]
```diff
@@ -84,10 +91,14 @@ protected override void RunScenario()
...
                         CatalogItemId = catalogItemId,
                         FriendlyName = "Myofferpurchase",
-                        Quantity = 1,
-                        TermDuration = (availability.Terms.First().Duration == null) ? null : availability.Terms.First().Duration,
+                        Quantity = termDuration == String.Empty ? 1 : 10,
+                        TermDuration = availability.Terms.First().Duration ?? null ,
                         BillingCycle = sku.SupportedBillingCycles.ToArray().First(),
-                        ProvisioningContext = ProvisioningContext
+                        ProvisioningContext = ProvisioningContext,
+                        RenewsTo = termDuration == String.Empty ? null : new RenewsTo()
```
Comment (SankarGopalan) [inline sdk/SdkSamples/Carts/CreateCart.cs]: Get RenewalTermDuration and Assign CartLineItem.RenewsTo only if (sku.IsTrial && availability.Terms.Any(r => r.RenewalOptions.Any(t => t.TermDuration != null)))

File: sdk/SdkSamples/App.config [hunk 2]
```diff
@@ -117,6 +117,8 @@
     <add key="DefaultBillingCycle" value="" />
     <!-- The Default Customers Agreement csv file, leave empty to prompt user to enter it -->
     <add key="DefaultCustomerAgreementCsvFileName" value="" />
+    <!-- The Default Renewal Term Duration, leave empty to prompt user to enter it -->
+    <add key="DefaultTermDuration" value="" />
```
Comment (SankarGopalan) [inline sdk/SdkSamples/App.config]: DefaultRenewalTermDuration?

File: sdk/SdkSamples/App.config [hunk 3]
```diff
@@ -117,6 +117,8 @@
     <add key="DefaultBillingCycle" value="" />
     <!-- The Default Customers Agreement csv file, leave empty to prompt user to enter it -->
     <add key="DefaultCustomerAgreementCsvFileName" value="" />
+    <!-- The Default Renewal Term Duration, leave empty to prompt user to enter it -->
```
Comment (SankarGopalan) [inline sdk/SdkSamples/App.config]: Update Subscriptions sample to display RenewalTermDuration.

File: sdk/SdkSamples/Carts/CreateCart.cs [hunk 4]
```diff
@@ -84,10 +91,14 @@ protected override void RunScenario()
                     {
                         CatalogItemId = catalogItemId,
                         FriendlyName = "Myofferpurchase",
-                        Quantity = 1,
-                        TermDuration = (availability.Terms.First().Duration == null) ? null : availability.Terms.First().Duration,
+                        Quantity = termDuration == String.Empty ? 1 : 10,
```
[the remark this answers is not in the document]
Comment (sujith0406) [reply sdk/SdkSamples/Carts/CreateCart.cs]: Fixed

File: sdk/SdkSamples/Carts/CreateCart.cs (hunk 1 above)
Comment (sujith0406) [reply sdk/SdkSamples/Carts/CreateCart.cs]: Fixed

File: sdk/SdkSamples/App.config (hunk 2 above)
Comment (sujith0406) [reply sdk/SdkSamples/App.config]: Fixed

File: sdk/SdkSamples/App.config (hunk 3 above)
Comment (sujith0406) [reply sdk/SdkSamples/App.config]: Not required as it is a get operation.

**Pull request merged**